packages/vaadin-themable-mixin/vaadin-theme-detection-mixin.js

```diff
@@ -0,0 +1,53 @@
+import { findRoot } from './lumo-injection-mixin.js';
+import { ThemeDetector } from './src/theme-detector.js';
+
+/**
+ * Mixin for detecting which Vaadin theme is applied to the application.
+ * Automatically adds a `vaadin-theme` attribute to the host element
+ * with the name of the detected theme (`lumo` or `aura`), which can be
+ * used in component styles to apply theme-specific styling.
+ */
+export const ThemeDetectionMixin = (superClass) =>
+  class VaadinThemeDetectionMixin extends superClass {
+    constructor() {
+      super();
+
+      this.__applyDetectedTheme = this.__applyDetectedTheme.bind(this);
+    }
+
+    connectedCallback() {
```

Comment:
nit: let's add `/** @protected */` JSDoc annotation to `connectedCallback()` and `disconnectedCallback()`.